File: daily/python/Nov-18-Min-depth-of-tree.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        res = self.recur(root)
        if res == 1e9:
            return 0
        else:
            return res

    def recur(self, curr: Optional[TreeNode]) -> int:
        if curr==None:
            return 1e9
        if curr.left==None and curr.right==None:
            return 1
        return 1 + min(self.recur(curr.left), self.recur(curr.right))

File: daily/cpp/Nov-18-Min-depth-of-tree.cpp

using namespace std;
struct TreeNode {
  int val;
  TreeNode *left;
  TreeNode *right;
  TreeNode() : val(0), left(nullptr), right(nullptr) {}
  TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
  TreeNode(int x, TreeNode *left, TreeNode *right)
      : val(x), left(left), right(right) {}
};

class Solution {
 public:
  int minDepth(TreeNode *root) {
    int res = recur(root);
    return res == INT_MAX ? 0 : res;
  }

  int recur(TreeNode *curr) {
    if (!curr) return INT_MAX;
    if (!curr->left && !curr->right) return 1;
    return 1 + min(recur(curr->left), recur(curr->right));
  }
}
